### ISSUE TITLE
Using automations with Fordpass_tracker

### ISSUE THREAD

**Issue by JVTEAM** [itchannel/fordpass-ha#207]
I've been attempting to create automations based on the tracker entering certain zones.   However when I use "Zone" and enters or leaves it never seems to trigger the automation.  Does anyone know the best way to run an automation on this?

**Comment by rtw819** [comment]
Having a similar issue with Zone based location checks.  I'm new to Automations, but I've been playing and added a Condition to a couple automations to see if the vehicle is at home, and it results in a failure condition every time.  Yes, there is a Zone defined for "home" (see logbook entries copied below).

YAML portion of the Automation Condition, based on other similar conditions...
```
condition: zone
entity_id: device_tracker.fordpass_tracker
zone: zone.home
```

Clicking the "Test" link button by the (above) Zone-based condition results in a pop-up window:
```
Error occurred while testing condition
Unknown error
```

Logbook shows:
```
fordpass_tracker was detected at home
9:14:46 PM - 3 hours ago
fordpass_tracker was detected at unavailable
9:09:45 PM - 3 hours ago
fordpass_tracker was detected at home
8:44:14 PM - 3 hours ago
fordpass_tracker was detected away
7:14:11 PM - 5 hours ago
```

home-assistant.log output:

```
2022-08-18 23:54:47.215 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547282534112] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 623, in handle_test_condition
    msg["id"], {"result": check_condition(hass, msg.get("variables"))}
  File "/usr/src/homeassistant/homeassistant/helpers/condition.py", line 145, in wrapper
    result = condition(hass, variables)
  File "/usr/src/homeassistant/homeassistant/helpers/condition.py", line 871, in if_in_zone
    if zone(hass, zone_entity_id, entity_id):
  File "/usr/src/homeassistant/homeassistant/helpers/condition.py", line 851, in zone
    return zone_cmp.in_zone(
  File "/usr/src/homeassistant/homeassistant/components/zone/__init__.py", line 154, in in_zone
    zone_dist = distance(
  File "/usr/src/homeassistant/homeassistant/util/location.py", line 70, in distance
    result = vincenty((lat1, lon1), (lat2, lon2))
  File "/usr/src/homeassistant/homeassistant/util/location.py", line 95, in vincenty
    U1 = math.atan((1 - FLATTENING) * math.tan(math.radians(point1[0])))
TypeError: must be real number, not str

```

.


Does your condition look similar?  Are we missing something obvious?  Are there other places we can look for more details?

**Comment by JVTEAM** [comment]
I've been able to get a little bit of progress on here. I can now trigger home and work but I can't get it to do away for some reason.

I use "State" as the trigger type then put in the fordpass_tracker as the entity.  I can then put "home" or "Work" in the "to" portion.  I did have an epiphany this morning that if I put all those locations in the "from" spot that perhaps it will trigger that way.

**Comment by itchannel** [comment]
So I use the state as the trigger for mine which works fine. I could never get the zone trigger to work on any of my tracker entities not just Fordpass. But using the state trigger and setting the To field to the zone you want normally works.

**Comment by woempiej** [comment]
> I've been attempting to create automations based on the tracker entering certain zones. However when I use "Zone" and enters or leaves it never seems to trigger the automation. Does anyone know the best way to run an automation on this?

Is it an option for you to use the HA person tracker for this? Me and my Wife use the HA app (installed on the mobile phone). I use many automations with this entities in combination with zones.

**Comment by woempiej** [comment]
![afbeelding](https://user-images.githubusercontent.com/62903344/185808139-39f09e60-e2c7-4de2-b7ed-a6353ae86fe9.png)

**Comment by JacobWasFramed** [comment]
> So I use the state as the trigger for mine which works fine. I could never get the zone trigger to work on any of my tracker entities not just Fordpass. But using the state trigger and setting the To field to the zone you want normally works.

Can confirm, had the same issue with zones, but utilizing a State condition type and the fordpass tracker entity works without issue for a while now.

**Comment by rtw819** [comment]
> Can confirm, had the same issue with zones, but utilizing a State condition type and the fordpass tracker entity works without issue for a while now.

Please pardon my ignorance, but what attribute and state are you using to accomplish this?

Here is an example that doesn't work (doesn't pass the test):
```
condition: state
entity_id: device_tracker.fordpass_tracker
attribute: status
state: zone.home
```

**Comment by JVTEAM** [comment]
> > Can confirm, had the same issue with zones, but utilizing a State condition type and the fordpass tracker entity works without issue for a while now.
> 
> Please pardon my ignorance, but what attribute and state are you using to accomplish this?
> 
> Here is an example that doesn't work (doesn't pass the test):
> 
> ```
> condition: state
> entity_id: device_tracker.fordpass_tracker
> attribute: status
> state: zone.home
> ```

trigger:
  - platform: state
    entity_id:
      - device_tracker.fordpass_tracker
    to: Work

**Comment by itchannel** [comment]
@rtw819 I think your issue is using the zone. prefix when using the state trigger you only need to give the zone name not the entity id e.g. "Work" instead of "zone.Work"

**Comment by rtw819** [comment]
Ah, I've been looking for a working Condition not a Trigger, that would seem to be the difference.

One of my use cases where I need the location/condition is:  I would like to make sure all the doors/windows are closed, and the doors are locked at 11pm -- but only if the vehicle state is "Home".  Every combination of things I've tried with the Condition option have not passed the validation checks.  I guess perhaps I need to figure out if I can somehow massage a Trigger into providing the same sort of logical function as an "at home" conditional response.  Thanks for the ideas thus far.

**Comment by itchannel** [comment]
@rtw819 This works for me

```
alias: fordpass_test1
description: ""
trigger:
  - platform: time
    at: "19:00:00"
condition:
  - condition: state
    entity_id: device_tracker.fordpass_tracker
    state: Home
  - condition: state
    entity_id: sensor.fordpass_doorstatus
    state: Open
action:
  - service: notify.persistent_notification
    data:
      message: Doors open
mode: single
```

**Comment by rtw819** [comment]
@itchannel  - Thank you for this...

It turns out that after I had manually added the simple YAML snippet, the dropdown for the Condition editor then allowed me to pick the correct "Home" value, which turned out to be "home" instead.  It seems the tracker location state is case sensitive somehow and doesn't pass the test if it isn't matched exactly, including case.

Here's what I ended up using (for the condition section):

```
condition: and
conditions:
  - condition: state
    entity_id: device_tracker.fordpass_tracker
    state: home
  - condition: state
    entity_id: sensor.fordpass_doorstatus
    state: Open
    for:
      hours: 0
      minutes: 10
      seconds: 0
```

Thanks again!!

**Comment by boldrikhun** [comment]
Hi!
I have the same issue, that I raised at [HA Forum](https://community.home-assistant.io/t/fordpass-location-based-automation-error/489159) as well (no response yet):

**So some additional information to this:**

- Studio Code Server says: “String does not match the pattern…”
- Automation condition test says: “Error occurred while testing condition: Unknown error”
- Automation Traces says: “Error: must be real number, not str”

There’re two options to get the GPS location of the car:
1. sensor.fordpass_gps
2. device_tracker.fordpass_tracker
but both gives this error.

Following the Traces error message, I found, that with both devices, the Longitude and Latitude Attributes are between quotes (so truly are strings, not like with other devices). But strangely, if I use the sensor or the device above for a dashboard card, it works perfect…

Anyways, as a workaround I'll try the methods you use till someone figures this out :)
![image](https://user-images.githubusercontent.com/118205153/202789788-8ef99872-7342-42d1-896f-cd7555c11ffc.png)
